### PR TITLE
`SpectraNet` model implementation

### PR DESCRIPTION
**Pull request by maxwest-uw**
Takes the `SpectraNet` and `SpectraNetRedshift` models and rolls them into a single hyrax model that can optionally switch the output between transient classification and redshift estimates.

- creates a number of default config parameters for the `SpectraNet` model
- adds a new basic spectra dataset that can be expanded at a later date but currently takes in an absolute path to a `.pt` file of preprocessed data .

### CODE REVIEW

File: AppleCider/hyrax_datasets/spectra_dataset.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import torch
+from hyrax.data_sets.data_set_registry import HyraxDataset
+from torch.utils.data import Dataset
+
+
+class SpectraData(HyraxDataset, Dataset):
+    def __init__(self, config, data_location=None):
+        super().__init__(config)
+
+        data_table = torch.load(config["model_inputs"]["data"]["file_path"])
```
Comment (drewoldag) [inline AppleCider/hyrax_datasets/spectra_dataset.py]: If you update the `max_config.toml` file to use `data_location`, then this line can become
```suggestion
        data_table = torch.load(data_location)
```

File: AppleCider/models/hyrax_models/spectranet.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        else:
+            self.classifier = nn.Sequential(
+                nn.Linear(flat_dim, 384),
+                nn.LayerNorm(384),
+                nn.GELU(),
+                nn.Dropout(0.5),
+                nn.Linear(384, class_order),
+            )
+
+    def forward(self, x):
```
Comment (drewoldag) [inline AppleCider/models/hyrax_models/spectranet.py]: I believe that `forward` will always receive data that has passed through `to_tensor` whether that is for training or inference. The difference is that `train_step` will call the `forward` method during training, but during inference, the data is fed directly from `to_tensor` to `forward`. 

Long preamble to say that forward should accept the batch as input, and then unpack it like what happens on line 183 in `train_step`, but it won't use `labels` or `redshifts`...

i.e. 
```
def forward(self, batch):
    inputs, _, _ = batch
    # keep the rest the same
```

File: AppleCider/hyrax_datasets/spectra_dataset.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        self._redshifts = data_table["redshifts"]
+        self._file_paths = data_table["file_paths"]
+
+        metadata_table = self._read_metadata()
+        self.table = metadata_table
+        super().__init__(config, metadata_table)
+
+    def ids(self):
+        """Return the ids of the data set"""
+        return np.arange(len(self._data))
```
[the remark this answers is not in the document]
Comment (maxwest-uw) [reply AppleCider/hyrax_datasets/spectra_dataset.py]: there isn't, at least not in the preprocessed data I was sent. maybe we can propagate that through once we get deeper into the processing pipeline?
